sklearn/multioutput.py

```diff
@@ -603,3 +607,125 @@ def decision_function(self, X):
         Y_decision = Y_decision_chain[:, inv_order]
 
         return Y_decision
+
+
+class RegressorChain(BaseChain, RegressorMixin):
+    """A multi-label model that arranges regressions into a chain.
+
+    Each model makes a prediction in the order specified by the chain using
+    all of the available features provided to the model plus the predictions
+    of models that are earlier in the chain.
+
+    Parameters
+    ----------
+    base_estimator : estimator
+        The base estimator from which the classifier chain is built.
+
+    order : array-like, shape=[n_outputs] or 'random', optional
+        By default the order will be determined by the order of columns in
+        the label matrix Y.::
+
+            order = [0, 1, 2, ..., Y.shape[1] - 1]
+
+        The order of the chain can be explicitly set by providing a list of
+        integers. For example, for a chain of length 5.::
+
+            order = [1, 3, 2, 4, 0]
+
+        means that the first model in the chain will make predictions for
+        column 1 in the Y matrix, the second model will make predictions
+        for column 3, etc.
+
+        If order is 'random' a random ordering will be used.
+
+    cv : int, cross-validation generator or an iterable, optional (
+    default=None)
+        Determines whether to use cross validated predictions or true
+        labels for the results of previous estimators in the chain.
+        If cv is None the true labels are used when fitting. Otherwise
+        possible inputs for cv are:
+            * integer, to specify the number of folds in a (Stratified)KFold,
+            * An object to be used as a cross-validation generator.
+            * An iterable yielding train, test splits.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+        The random number generator is used to generate random chain orders.
+
+    Attributes
+    ----------
+    estimators_ : list
+        A list of clones of base_estimator.
+
+    order_ : list
+        The order of labels in the classifier chain.
+
+    """
+
+    def fit(self, X, Y):
```

Comment:
This method is almost **entirely** duplicated in `BaseChain`, the only difference is `self.classes_` that concerns only `ClassifierChain`.

I suggest to remove the `self.classes_` from `BaseChain.fit`, in order to have in `BaseChain` the shared code.
Then I would remove `fit` from `RegressorChain`, and would have the following in `ClassifierChain`:

```py
    def fit(self, X, Y):
       """Fit the model to data matrix X and targets Y.
         Parameters
        ----------
        X : {array-like, sparse matrix}, shape (n_samples, n_features)
            The input data.
        Y : array-like, shape (n_samples, n_classes)
            The target values.
         Returns
        -------
        self : object
            Returns self.
        """
        super(ClassifierChain, self).fit(X, Y)
        self.classes_ = []
        for chain_idx, estimator in enumerate(self.estimators_):
            self.classes_.append(estimator.classes_)
        return self
```